### PR TITLE
feat(company role): support optional terms

### PR DESCRIPTION
**Pull request by lavanya-bmw**
## Description

enabled agreement handling by adding frontend logic and UI handling of optional agreements

## Why

To support optional terms for company roles

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/584

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [ ] I have added tests that prove my changes work

### CODE REVIEW

**Comment by evegufy** [comment]
@lavanya-bmw regarding the failing dependencies check: I noticed a duplication issue with the following two IP issues
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15317#note_2470453
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/15286#note_2470455
I signal that to the IP Team, let's wait a couple of hours for the response

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,8 @@
 - Company Application Registraion Detail Overlay
   - UI improvements
   - Add correct registration application checklist items
+- Company Role
+  - enabled agreement handling by adding frontend logic and UI handling of optional agreements
```
Comment (evegufy) [inline CHANGELOG.md:23]: you added this change to the bugfix section even though the PR title and issue #584 suggest feature, what is correct?

**Review by evegufy — CHANGES_REQUESTED** [review]
could you please clarify the milestone of https://github.com/eclipse-tractusx/portal-frontend/issues/584? Currently it's still set to our latest past release (2.0.0), should it be the 2.1.0 release (release/v2.1.0-RC1 branch) or the release after (dev branch)?

File: CHANGELOG.md (hunk 1 above)
Comment (lavanya-bmw) [reply CHANGELOG.md]: @evegufy updated changelog to feature and regarding milestone, I checked with Julia and confirmed to create PR against release/v2.1.0-RC1

**Review by evegufy — APPROVED** [review]

**Pull request merged**